src/tcp.c

```diff
@@ -32,6 +32,24 @@
 
 #define get_sin_port(addr) (((struct sockaddr_in*) addr->ai_addr)->sin_port)
 
+static struct enftun_tcp_ops enftun_tcp_native_ops = {
+    .connect = (int (*)(void*, const char* host, const char*))
+        enftun_tcp_native_connect,
+    .connect_any = (int (*)(void*, const char** host, const char*))
+        enftun_tcp_native_connect_any,
+    .close = (void (*)(void*)) enftun_tcp_native_close};
+
+void
+enftun_tcp_native_init(struct enftun_tcp_native* ctx,
+                       struct enftun_tcp* sock,
```

Comment:
Remove the `sock` argument here.  Just set `sock = &sock_native.base` after calling `enftun_tcp_native_init` in `tls.c` or `xtt.c`.

Doing that here is confusing --- the init function becomes responses for both init-ing (good) and setting some pointer passed in by the caller. What if the caller doesn't have an `struct enftun_tcp` pointer to set?  Better to let the caller do it themselves, if desired.